test/JDBC/expected/BABEL_4360.out

```diff
@@ -0,0 +1,159 @@
+DECLARE @a TABLE (a INT);
+INSERT INTO @a EXECUTE('SELECT * FROM t1; SELECT 3');
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: relation "t1" does not exist)~~
+
+
+
+-- tsql
+
+-- Test nested procedure calls with rollback and error case
+CREATE TABLE babel_4360_t (id INT);
+GO
+
+-- psql     currentSchema=master_dbo,public
+
+CREATE PROCEDURE psql_interop_proc1()
+AS
+$$
+BEGIN
+  ROLLBACK;
+  INSERT INTO babel_4360_t VALUES (99);
+END
+$$ LANGUAGE PLPGSQL;
+GO
+
+CREATE PROCEDURE psql_interop_proc2()
+AS
+$$
+BEGIN
+  CALL psql_interop_proc1();
+  INSERT INTO babel_4360_t VALUES (98);
+END
+$$ LANGUAGE PLPGSQL;
+GO
+
+-- tsql
+
+CREATE PROCEDURE tsql_interop_proc1
+AS
+EXEC psql_interop_proc2;
+INSERT INTO babel_4360_t VALUES (97);
+GO
+
+CREATE PROCEDURE tsql_interop_proc2
+AS
+EXEC tsql_interop_proc1;
+INSERT INTO babel_4360_t VALUES (96);
+GO
+
+-- psql     currentSchema=master_dbo,public
+
+CREATE PROCEDURE psql_interop_proc3()
+AS
+$$
+BEGIN
+  CALL tsql_interop_proc2();
+  INSERT INTO babel_4360_t VALUES (95);
+END
+$$ LANGUAGE PLPGSQL;
+GO
+
+CREATE PROCEDURE psql_interop_proc4()
+AS
+$$
+BEGIN
+  CALL psql_interop_proc3();
+  INSERT INTO babel_4360_t VALUES (94);
+END
+$$ LANGUAGE PLPGSQL;
+GO
+
+-- tsql
+
+CREATE PROCEDURE tsql_interop_proc3
+AS
+EXEC psql_interop_proc4;
+INSERT INTO babel_4360_t VALUES (93);
+GO
+
+CREATE PROCEDURE tsql_interop_proc4
+AS
+EXEC tsql_interop_proc3;
+INSERT INTO babel_4360_t VALUES (92);
+GO
+
+EXEC tsql_interop_proc4;
```

Comment:
what happens without the fix? do we have test cases for both error and leak to validate fix?

Reply:
insert into @tablevariable execute ('some error condition') --> SPI stack depth expected > SPI stack current (error scenario)

Difficult to test leak, since abort transaction will clean up everything.